### PR TITLE
Cet-4385 backstage initiative page ladder driven UI

### PR DESCRIPTION
**Pull request by igorRoog**
## Change description

> Description here

Revamp of Initiative page for both ladder- and score-driven.

image
image
image
image
image

### CODE REVIEW

File: src/components/Common/LinearProgressWithLabel.tsx [hunk 1]
```diff
@@ -71,7 +72,11 @@ export const LinearProgressWithLabel: React.FC<LinearProgressWithLabelProps> =
           />
         </Box>
         <Box minWidth={45}>
-          <Typography variant="body1">{`${Math.round(value)}%`}</Typography>
+          {label ? (
+            <Typography variant="body1">{label}</Typography>
+          ) : (
+            <Typography variant="body1">{`${Math.round(value)}%`}</Typography>
+          )}
```
Comment (FoodProduct) [inline src/components/Common/LinearProgressWithLabel.tsx:79]: nit: why not
```suggestion
          <Typography variant="body1">{label ?? `${Math.round(value)}%`}</Typography>
```
?

File: src/components/Initiatives/InitiativeDetailsPage/InitativeFilterDialog/InitiativeFilterDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+          />
+        )}
+      </DialogContent>
+      <DialogActions>
+        <Button
+          onClick={handleSaveFilters}
+          color="primary"
+          aria-label="Close modal"
+        >
+          Close
```
Comment (FoodProduct) [inline src/components/Initiatives/InitiativeDetailsPage/InitativeFilterDialog/InitiativeFilterDialog.tsx:168]: Seems weird that this is labeled "close" but seems to really be "save" or "apply filters" -- could we update the copy?

File: src/components/Initiatives/InitiativeDetailsPage/InitativeFilterDialog/InitiativeFilterDialogUtils.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    formatProperty: (groupRef: string) =>
+      humanizeEntityRef(parseEntityRef(groupRef), defaultSystemRefContext),
+  },
+];
+
+export const toPredicateFilters = (
+  filters: { [id: string]: boolean },
+  name: string,
+) => {
+  const predicateFilters: Record<string, boolean> = {};
```
Comment (FoodProduct) [inline src/components/Initiatives/InitiativeDetailsPage/InitativeFilterDialog/InitiativeFilterDialogUtils.tsx:63]: [Non-blocking] could have been a `reduce`

File: src/components/Initiatives/InitiativeDetailsPage/InitiativeDetailsPage.tsx [hunk 4]
```diff
@@ -55,48 +63,140 @@ export const InitiativeDetailsPage = () => {
...
+          </Button>
+        )}
+      </Box>
+      <Box my={1}>{renderTab(activeTab)}</Box>
+      <InitiativeFilterDialog
+        isOpen={isFilterDialogOpen}
+        actionItems={actionItems}
+        initiative={initiative}
+        handleClose={handleFilterDialogClose}
+        setFilter={(filter: any) => setFilter(() => filter)}
```
Comment (FoodProduct) [inline src/components/Initiatives/InitiativeDetailsPage/InitiativeDetailsPage.tsx:198]: [Non-blocking] Is there a type we can use that's more descriptive than `any` here?

File: src/components/Initiatives/InitiativeDetailsPage/InitiativeFailingTab/InitiativeFailingTab.tsx [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+          tag,
+          score: (numRules - serviceActionItems.length) / numRules,
+        };
+      })
+      .sort((left, right) => left.actionItems.length - right.actionItems.length)
+      .sort((left, right) => left.tag.localeCompare(right.tag));
+  }, [entitiesByTag, failingComponents, numRules]);
+
+  const showPagination = useMemo(() => {
+    return Object.keys(failingComponents).length > defaultPageSize;
```
Comment (FoodProduct) [inline src/components/Initiatives/InitiativeDetailsPage/InitiativeFailingTab/InitiativeFailingTab.tsx:87]: nit: can use `_.size`
```suggestion
    return size(failingComponents) > defaultPageSize;
```

File: src/components/Initiatives/InitiativeDetailsPage/InitiativeFailingTab/InitiativeFailingTab.tsx [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+          <Typography variant="subtitle2" className={classes.tag}>
+            {componentRef}
+          </Typography>
+        ),
+      },
+      {
+        field: 'score',
+        sorting: true,
+        title: 'Score',
+        width: '20%',
```
Comment (FoodProduct) [inline src/components/Initiatives/InitiativeDetailsPage/InitiativeFailingTab/InitiativeFailingTab.tsx:150]: The width percentages add up to 120% 🧐

File: src/components/Initiatives/InitiativeDetailsPage/InitiativeFailingTab/ServiceNameAndRulesColumn.tsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+              <Typography variant="subtitle2">{description}</Typography>
+            )}
+          </Box>
+        </Box>
+        <Box display="flex" flexDirection="column">
+          {actionItems.map(actionItem => {
+            return (
+              <ScorecardServiceRuleRow
+                hideWeight
+                key={`${componentRef}-FailingRule-${actionItem.rule.id}`}
```
Comment (FoodProduct) [inline src/components/Initiatives/InitiativeDetailsPage/InitiativeFailingTab/ServiceNameAndRulesColumn.tsx:55]: Feels weird for a `*Ref` variable to be a string / included in the key 🤔

File: src/components/Initiatives/InitiativeDetailsPage/InitiativePassingTab/InitiativePassingTab.tsx [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+          <Typography variant="subtitle2" className={classes.tag}>
+            {componentRef}
+          </Typography>
+        ),
+      },
+      {
+        field: 'score',
+        sorting: false,
+        title: 'Score',
+        width: '20%',
```
Comment (FoodProduct) [inline src/components/Initiatives/InitiativeDetailsPage/InitiativePassingTab/InitiativePassingTab.tsx:149]: same re: width adds up to 120%

File: src/components/Initiatives/InitiativeDetailsPage/InitiativeStatsCard/InitiativeStatsCard.tsx [hunk 9]
```diff
@@ -67,17 +68,17 @@ export const InitiativeStatsCard = ({
...
-          label: 'Complete',
-          status: percentageToStatus(complete),
-          value: percentify(complete),
-          type: 'PERCENTAGE',
-        },
-        { label: 'Passing', status: Status.OKAY, value: numPassing },
-        { label: 'Failing', status: Status.ERROR, value: numFailing },
-      ]}
-    />
+    <Paper elevation={0} style={{ marginBottom: 8 }}>
```
Comment (FoodProduct) [inline src/components/Initiatives/InitiativeDetailsPage/InitiativeStatsCard/InitiativeStatsCard.tsx:71]: TIL

File: src/components/Scorecards/ScorecardDetailsPage/ScorecardFilterDialog/ScorecardFilterDialog.tsx [hunk 10]
```diff
@@ -84,25 +115,31 @@ export const ScorecardFilterDialog = ({
...
+    },
+    {
+      name: 'Failing rules',
       filters: ruleFilterDefinitions,
       generatePredicate: (failingRule: string) =>
         createApplicableRulePredicate(false, failingRule),
     },
     {
-      name: 'Passing Rule',
+      name: 'Passing rules',
```
Comment (FoodProduct) [inline src/components/Scorecards/ScorecardDetailsPage/ScorecardFilterDialog/ScorecardFilterDialog.tsx:130]: 🙏

**Review by FoodProduct — COMMENTED** [review]
* Overall the code looks very clean!
* I did not functionally test locally
* I recognize that some of the code changes might be moved code and I may have made some nits on existing moved code -- feel free to dismiss those if they're annoying to update
* It's a very big PR so I'm sure I missed things regardless

Are there are any specific areas you'd like me to focus code review and/or anything you'd like help functionally testing?

**Comment by igorRoog** [comment]
> * Overall the code looks very clean!
> * I did not functionally test locally
> * I recognize that some of the code changes might be moved code and I may have made some nits on existing moved code -- feel free to dismiss those if they're annoying to update
> * It's a very big PR so I'm sure I missed things regardless
> 
> Are there are any specific areas you'd like me to focus code review and/or anything you'd like help functionally testing?

@FoodProduct Thanks for reviewing this PR! I'll try to fix all the spotted nits, regardless if given code is new or just moved ;). Actually, I can't think about any "hot" areas which should be tested more intensively. And when it comes to the testing - we already tested it together with Cristina, but if you'll find some time to create one or two scorecards with some initiatives/rules exemptions etc and play around it in backstage, it would be great!

File: src/components/Initiatives/InitiativeDetailsPage/InitiativeFailingTab/InitiativeFailingTab.tsx (hunk 5 above)
Comment (igorRoog) [reply src/components/Initiatives/InitiativeDetailsPage/InitiativeFailingTab/InitiativeFailingTab.tsx:87]: TIL ;)

File: src/components/Initiatives/InitiativeDetailsPage/InitiativeFailingTab/ServiceNameAndRulesColumn.tsx (hunk 7 above)
Comment (igorRoog) [reply src/components/Initiatives/InitiativeDetailsPage/InitiativeFailingTab/ServiceNameAndRulesColumn.tsx:55]: Oh, that's something specific for backstage I guess (and it is NOT react ref). You can search for "componentRef" in a code and that's basically a TAG, used widely across the app (and it is a string value indeed). So in that case I won't rename it.

Comment (FoodProduct) [reply src/components/Initiatives/InitiativeDetailsPage/InitiativeFailingTab/ServiceNameAndRulesColumn.tsx:55]: Ahhh interesting -- that rings a bell so I'm sure I noticed that before a while ago and forgot 😅

**Review by FoodProduct — APPROVED** [review]
LGTM with the same caveat that I didn't functionally test locally but I think you and Cristina did so!

**Pull request merged**